scripts/docker-integration-tests/prometheus/test.sh

```diff
@@ -189,6 +189,18 @@ function test_query_limits_applied {
   echo "Test query limit with coordinator limit header"
   ATTEMPTS=50 TIMEOUT=2 MAX_TIMEOUT=4 retry_with_backoff  \
     '[[ $(curl -s -H "M3-Limit-Max-Series: 10" 0.0.0.0:7201/api/v1/query?query=\\{__name__!=\"\"\\} | jq -r ".data.result | length") -eq 10 ]]'
+  
+  echo "Test query limit with require-exhaustive headers false (above)"
+  ATTEMPTS=50 TIMEOUT=2 MAX_TIMEOUT=4 retry_with_backoff  \
+    '[[ $(curl -s -H "M3-Limit-Max-Series: 10" -H "M3-Limit-Require-Exhaustive: false" 0.0.0.0:7201/api/v1/query?query=\\{__name__!=\"\"\\} | jq -r ".data.result | length") -eq 10 ]]'
+
+  echo "Test query limit with require-exhaustive headers true (below limit therefore no error)"
+  ATTEMPTS=50 TIMEOUT=2 MAX_TIMEOUT=4 retry_with_backoff  \
+    '[[ $(curl -s -H "M3-Limit-Max-Series: 101" -H "M3-Limit-Require-Exhaustive: true" 0.0.0.0:7201/api/v1/query?query=\\{__name__!=\"\"\\} | jq -r ".data.result | length") -eq 101 ]]'  
+  
+  echo "Test query limit with require-exhaustive headers true (above limit therefore error)"
+  expected_status=$(curl -s -H "M3-Limit-Max-Series: 10" -H "M3-Limit-Require-Exhaustive: true" 0.0.0.0:7201/api/v1/query?query=\\{__name__!=\"\"\\} | jq -r ".status")
+  test "$expected_status" = "error"
```

Comment:
Should we verify the specifics of the error here more? Only downside is if we change the msg then this needs to change as well.

Reply:
Yeah, I think these tests end up being pretty prescriptive, probably better to check here and force an update if necessary in the future (especially since we're keying on the exact type of error)